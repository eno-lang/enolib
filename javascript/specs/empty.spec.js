const { parse } = require('..');

const sample = `
empty:
leer:

-

-

-

nichts:
nada:

# nothing
## none
void:
emptyness:
-- leere
-- leere

# ningun
absence:
-

-
-


non:
end:
`;

describe('Empty elements', () => {
  it('correctly parses', () => {
    const doc = parse(sample);
    expect(doc.raw()).toMatchSnapshot();
  });
});
