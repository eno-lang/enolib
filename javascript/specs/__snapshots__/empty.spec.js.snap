// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty elements correctly parses 1`] = `
Array [
  Object {
    "empty": null,
  },
  Object {
    "leer": Array [
      null,
      null,
      null,
    ],
  },
  Object {
    "nichts": null,
  },
  Object {
    "nada": null,
  },
  Object {
    "nothing": Array [
      Object {
        "none": Array [
          Object {
            "void": null,
          },
          Object {
            "emptyness": null,
          },
          Object {
            "leere": null,
          },
        ],
      },
    ],
  },
  Object {
    "ningun": Array [
      Object {
        "absence": Array [
          null,
          null,
          null,
        ],
      },
      Object {
        "non": null,
      },
      Object {
        "end": null,
      },
    ],
  },
]
`;
